### PR TITLE
binding for string.endsWith

### PR DESCRIPTION
**Pull request by gaurikholkar**
Part of #275

### CODE REVIEW

File: tests/all/js_globals/JsString.rs [hunk 1]
```diff
@@ -627,3 +627,33 @@ fn value_of() {
...
+            }
+        "#)
+        .file("test.ts", r#"
+            import * as assert from "assert";
+            import * as wasm from "./out";
+
+            export function test() {
+                let str = "To be, or not to be, that is the question.";
+
+                assert.ok(wasm.string_ends_with(str, 'question', 0));
```
Comment (belfz) [inline tests/all/js_globals/JsString.rs]: You have a bug here. This should rather read `assert.ok(wasm.string_ends_with(str, 'question.', str.length));` in order to pass (notice the dot at the end of search string).

File: tests/all/js_globals/JsString.rs [hunk 2]
```diff
@@ -627,3 +627,33 @@ fn value_of() {
...
+            pub fn string_ends_with(this: &js::JsString, search_string: &js::JsString, length: u32) -> bool {
+                this.ends_with(search_string, length)
+            }
+        "#)
+        .file("test.ts", r#"
+            import * as assert from "assert";
+            import * as wasm from "./out";
+
+            export function test() {
+                let str = "To be, or not to be, that is the question.";
```
Comment (belfz) [inline tests/all/js_globals/JsString.rs]: This is minor, but please favor use of `const`, unless `let` is really needed.

File: tests/all/js_globals/JsString.rs (hunk 1 above)
Comment (belfz) [reply tests/all/js_globals/JsString.rs]: After you correct that, please make sure to run the tests with `cargo test js_globals::JsString` :)

**Comment by alexcrichton** [comment]
A good question! I'm not sure an `Option` will work in the sense that it'd have to pass `undefined` as the argument value, but another option here is something where we expose both versions of the function under different names. For example we could have:

```rust
#[wasm_bindgen(method, js_name = endsWith)]
fn ends_with(this: &JsString, tail: &JsString) -> bool;
#[wasm_bindgen(method, js_name = endsWith)]
fn ends_with_len(this: &JsString, tail: &JsString, length: u32) -> bool;
```

**Comment by belfz** [comment]
@alexcrichton I was thinking that passing `None` could be mapped to `undefined` when crossing the wasm/js boundaries. Although now that I am thinking about it again, maybe we need a `js::Undefined` type, which would act as a "subtype" of all other types?

 While what you proposed could indeed work, I'm afraid it would become too verbose, especially in case of functions/methods that accept more than one optional parameter. Plus, we have to remember, that in such cases we could still want such functions to be called with `undefined` value for arguments in the middle.

**Comment by alexcrichton** [comment]
Indeed it may not be generally applicable! I think for something like `ends_with` though it may be the most ergonomic though

**Comment by sendilkumarn** [comment]
Oops sorry @gaurikholkar Closing this PR as I think it is already merged into master.

**Pull request closed without merging**